src/MyEventEmitter.js

```diff
@@ -1,12 +1,200 @@
+/* eslint-disable no-console */
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.events = {};
+  };
+
+  validateInput(eventName = undefined, listener = undefined) {
+    return {
+      isNameValid: typeof eventName === 'string',
+      isListenerValid: typeof listener === 'function',
+    };
+  }
+
+  on(eventName, listener) {
+    const { events, validateInput } = this;
+    const { isNameValid, isListenerValid } = validateInput(eventName, listener);
+
+    if (!isNameValid && !isListenerValid) {
+      console.log(
+        'Invalid arguments. (eventName => string, listener => function)',
+      );
+
+      return;
+    };
+
+    if (!events[eventName]) {
+      events[eventName] = [];
+    }
+
+    events[eventName].push(listener);
+  };
+
+  once(eventName, listener) {
+    const { validateInput } = this;
+    const { isNameValid, isListenerValid } = validateInput(eventName, listener);
+
+    if (!isNameValid && !isListenerValid) {
+      console.log(
+        'Invalid arguments. (eventName => string, listener => function)'
+      );
+
+      return;
+    };
+
+    const wrapper = (...args) => {
+      listener(...args);
+      this.off(eventName, wrapper);
+    };
+
+    this.on(eventName, wrapper);
+  };
+
+  off(eventName, listener) {
+    const { events, validateInput } = this;
+    const { isNameValid, isListenerValid } = validateInput(eventName, listener);
+
+    if (!isNameValid && !isListenerValid) {
+      console.log(
+        'Invalid arguments. (eventName => string, listener => function)'
+      );
+
+      return;
+    };
+
+    events[eventName] = events[eventName]
+      .filter((removedListener) => removedListener !== listener);
+  };
+
+  emit(eventName, ...args) {
+    const { events, validateInput } = this;
+    const { isNameValid } = validateInput(eventName);
+
+    if (!isNameValid) {
+      console.log(
+        'Invalid argument. (eventName => string)'
+      );
+
+      return;
+    };
+
+    if (!events[eventName]) {
+      console.log('Event does not exist');
+
+      return;
+    }
+
+    events[eventName].forEach(listener => listener(...args));
+  };
+
+  prependListener(eventName, listener) {
+    const { events, validateInput } = this;
+    const {
+      isListenerValid,
+      isNameValid,
+    } = validateInput(eventName, listener).bothArgsAreValid;
+
+    if (!isNameValid && !isListenerValid) {
+      console.log(
```

Comment:
Use '||' instead of '&&' for checking invalid input: if (!isNameValid || !isListenerValid) { ... }